base/src/main/java/io/spine/query/package-info.java

```diff
@@ -19,7 +19,386 @@
+ *
+ *    Project.Status status = 3 [(column) = true];
+ *
+ *    UserView assignee = 4;
+ *
+ *    int32 days_since_started = 5 [(column) = true];
+ *
+ *    bool was_reassigned = 6 [(column) = true];
+ *
+ *    repeated string tag = 7;
+ * }
+ * </pre>
+ *
+ * <p>The {@code ProjectView} message is declared as a state of a future Projection. As it is
+ * a part of a Ubiquitous Language, its Protobuf definition serves as a contract for everyone
+ * wishing to interact with it. In particular, this message marks some of its fields
+ * with the {@code (column)} option. It tells everyone, that the instances of this Projection
+ * will be available for querying by the values of these fields.
+ *
+ * <p>In terms of the code generation, the framework automatically appends the corresponding
+ * Java type with the nested {@code Query} and {@code QueryBuilder} types. They are composed
+ * at build-time and contain the definitions of the columns for the respective storage record,
+ * and the query DSL.
+ *
+ * <p>Notice the difference with the {@link io.spine.query.RecordQueryBuilder RecordQueryBuilder}
+ * API, as the developer no longer needs to declare any types by hand.
+ *
+ * <p>Here is an example:
+ *
+ * <pre>
+ *     ProjectView.Query query = ProjectView
+ *        .newQuery()
+ *        .status().isNot(Status.CREATED)
+ *        .daysSinceStarted().isGreaterThan(5)
+ *        .build();
+ * </pre>
+ *
+ * <p>It produces a query for all the {@code ProjectView} records which status is not
+ * {@code CREATED} and which were started more than five days ago.
+ *
+ * <p>Disjunction (or OR statements) are also supported:</p>
+ *
+ * <pre>
+ *    {@literal Either<ProjectView.QueryBuilder> startedMoreThanMonthAgo = }
+ *        project -> project.daysSinceStarted().isGreaterThan(30);
+ *    {@literal Either<ProjectView.QueryBuilder> isDone =                  }
+ *        project -> project.status().is(Status.DONE);
+ *
+ * ProjectView.Query query =
+ *        ProjectView.newQuery()
+ *                   .either(startedMoreThanMonthAgo, isDone)
+ *                   .build();
+ * </pre>
+ *
+ * <p>This piece produces a query targeting the projects which are either done or started more
+ * than thirty days ago.
+ *
+ * <p>The first field of the Message is treated as an identifier. It's important to understand that
+ * Spine treats the field declared first in the order of reading. And not the one with
+ * the least index value.
+ *
+ * <p>The name of the ID field is preserved and is exposed for querying:
+ *
+ * <pre>
+ *     ProjectView.Query query = ProjectView.newQuery()
+ *                                          .project().is(expectedId)
+ *                                          .build();
+ * </pre>
+ *
+ * <h4>Custom Columns</h4>
+ *
+ * <p>In addition to the columns declared in the Entity state, the query API allows declaring
+ * the custom columns. They are intended to handle the cases in which some computed on-the-fly data
```

Comment:
> the custom columns

”the” is probably not needed here.

Reply:
@alexander-yevsyukov I think it should be there, as we speak of a _particular_ _kind_ of cases, right after "in which".